execute/observation.go

```diff
@@ -220,6 +297,27 @@ func (p *Plugin) getFilterObservation(
 		}
 	}
 
+	// Remove cursed requests.
+	{
+		ci, err := p.getCurseInfo(ctx)
```

Comment:
This one is technically valid because it's only observing the destination. But per our conversation, I've removed it anyway to simplify curse race handling. Any curses during the state machine would discard the invalid report during ShouldTransmit.